models.rb

```diff
@@ -106,7 +110,17 @@ def download_scores
       return
     end
 
-    update_scores(updated)
+    rank.nil? ? update_scores(updated) : updated.select { |score| !IGNORED_PLAYERS.include?(score['user_name']) }.uniq { |score| score['user_name'] }[rank]
```

Comment:
The reason is that we need the replay ID of the run, which is not saved to the database, so we have to redownload the scores for it. I could've also done another method for this, but since it could be achieved by just modifying this line I decided to do that. Would you rather have another method?

Reply:
I think I prefer another method. I've written the start of this comment a few times and deleted it, so I'm probably overthinking this given it's like a five-line method, but here are some thoughts in no particular order:

- This feels like another instance of 'if parameter is passed, do fairly-different operation', which you've fixed in a couple places (and those fixes are great, thanks). This isn't really 'downloading the scores' if the rank is passed in - it's mostly retrieving the *replay info* for a given rank.
- This method was already slightly poorly-named - it's actually *updating* the saved scores for a given level/episode from the latest on the server. So I think this should probably be named `update_scores` (which requires renaming `update_scores` - but that should probably be named `save_scores` anyways!). And that makes it even more clear that this is *not* doing the same thing.
- You could share code here between `update_scores` and `get_replay` (or whatever you want to call the other method...) for sure - I would honestly not be terribly upset to see this duplicated though, it's a single function call and an error message, so there's not much shame in just copying it ;) you might even want the error message to be different for each case, so there's an argument here for these methods being similar-but-separate.

So yeah. I think too much about minor things like this I'm pretty sure :P But I'd prefer this be separated overall.

Reply:
Alrighty, separate methods it is. It is indeed clearer this way. I also updated the names of the methods you mentioned in all the pertinent places, I'll do some more testing later to make sure nothing was broken in the process :P